### DISCUSSION TITLE
What is MEI Basic

### DISCUSSION THREAD

**Discussion by rettinghaus** [music-encoding/music-encoding#1467] [General]
I'm starting to wonder what MEI Basic is supposed to be and where it should be used. 

For example, people are wondering if [coloring musical symbols](https://github.com/music-encoding/music-encoding/pull/1445#issuecomment-2082399364) is within the "Basic" scope of notation. I would say yes, because it is often used for [educational purposes](https://en.wikipedia.org/wiki/Colored_music_notation). How more _basic_ can musical notation get? 

Is MEI Basic thought to be a replacement for MusicXML as an interchange format? Well, it fails completely in supporting unpitched music (see [this issue](https://github.com/musescore/MuseScore/issues/22519)). Does a triangle fall into the area of _advanced_ notation? 

What is MEI Basic used for? Are there any real life use cases where MEI Basic is currently in use? I know it is in use for offering MEI support in MuseScore. But there are some weird features available, like exporting dotted hairpins and dashed volta brackets. Does this still counts as _basic_ notation? More common notational features like printing symbols smaller (except for grace notes) is not possible.

The Guidelines state MEI Basic 

> provides a simplified subset of the MEI framework that reflects the capabilities of most popular "Common Western Music Notation" score-writing applications currently in use.

From what I see it still misses this goal by miles. MEI Basic is very inconsistent in its feature set.

**Comment by ahankinson** [comment]
It is a work-in-progress, and the intention is to see what features are *actually* used and modify it to meet those needs. This is approaching it from a slightly different angle from MEI itself, which is "imagine all the things that might happen, and then figure out a way to encode them." Both approaches are valid.

> Well, it fails completely in supporting unpitched music (see https://github.com/musescore/MuseScore/issues/22519). Does a triangle fall into the area of advanced notation?

While un-pitched notes are not necessarily uncommon, they're also not an everyday occurrence. I'm sure it will get fixed. (The issue is only a week old, and we have lots of other things that need our attention too...)

> Are there any real life use cases where MEI Basic is currently in use? 

https://musescore.org/en

https://verovio.org

Do you need anything else for a release that's less than a year old?

> From what I see it still misses this goal by miles

I'm sure you can appreciate that software and specifications can take time to get right. Personally, I *don't* think it misses this mark by miles, since MEI Basic is what allowed us to create the MuseScore export in the first place. It was a necessary first step.

I'm sure you have it very clear in your head what a perfect world looks like. The rest of us are doing our best to meet your high expectations. Please be patient.

**Comment by rettinghaus** [comment]
Please pardon my polemics. But I would like to initiate an active discussion in the community about where we want to go with MEI Basic.

**Reply by ahankinson** [reply]
Your "polemics" can hurt those who have actually worked quite hard on this, and see it as a positive step forward for the MEI community as we seek to expand support for MEI in new places.

**Comment by craigsapp** [comment]
I am also wondering what MEI Basic is for (I use verovio MEI): https://github.com/music-encoding/music-encoding/pull/1445#issuecomment-2077627148

There should be clear guidelines on what is and is not the scope of MEI Basic in terms of elements and attributes.   It most likely seems it should be either the intersection or union of elements/attributes that are handled by MuseScore direct MEI export and SibMEI.  Or if specifically for educational purposes, then there would be a very limited set of elements and attributes.  And possibly what elements/attributes that should be implemented in an MEI-to-MusicXML exporter.

**Reply by ahankinson** [reply]
What is, and what isn't, in scope is intended to be worked out through usage. Getting a first release out the door is always hard to get right, and we could have spent many more years trying to "perfect" it, all the while losing steam and guaranteeing nothing happened. As it is, "MEI Basic" was first worked out at the MEC in Montreal (2014) and didn't actually get released until 2023. At a certain point it needed to just "get out there", even though we knew there were things that were missing.

Those who are not sure would do well to read the section in the Guidelines that cover MEI Basic: https://music-encoding.org/guidelines/dev/content/introduction.html#meiBasic

**Reply by kepper** [reply]
There is also a paragraph inside https://music-encoding.org/guidelines/dev/content/introduction.html#meiProfiles to consider. But in general, it is a good thing if the community picks this up and starts to discuss MEI Basic – we just need to do it carefully. Clearly, nNot everything done so far was right, but most of it wasn't wrong either…

**Comment by craigsapp** [comment]
> While un-pitched notes are not necessarily uncommon, they're also not an everyday occurrence. 

It is very basic to a triangle performers...  MusicXML has/had a basic set designed for Hal Leonard, which I think is proprietary, and covered what features they needed to do on-demand printing of mostly popular music parts in music stores. Initially it did not include C clefs, but that caused problems for printing viola music.

**Comment by craigsapp** [comment]
There is also this announcement which is pretty clear on the intended scope:

https://music-encoding.org/update/2019/05/31/mei-basic-first-draft.html

> The MEI Community is pleased to announce the first draft version of MEI Basic, a simplified profile of MEI intended to support data interchange both within MEI and with other encoding formats.
> 
> MEI Basic has been discussed at several occasions over the past three years. It intends to simplify MEI so that every feature can only be expressed in one specific way, and also removes all editorial or analytical possibilities, as well as the integration of facsimiles and recordings. What is left is a small core of MEI’s features, that falls in the range of other encoding formats and thus allows an easier transformation.  This would also limit things such as `<accid>` as a child of `<note>` (preventing editorial accidentals), since `note@accid` and `note/accid` are two different ways to expressing the same thing (in simple cases).

Since there is now direct export of MEI (Basic) from MuseScore, it seems that anything that is capable of the MEI exporter in MuseScore would be the baseline for MEI Basic features.  (and preferably import back into MuseScore or into another music editor either directly through MEI or MusicXML).

I would presume that things like `<beamSpan>` and `<tupletSpan> ` would be excluded since they are alternate expressions of `<beam>` and `<tuplet>`, but also that would limit certain repertories of music (which is not necessarily a bad thing, which can use full MEI and likely have difficulties with export/import with typical music editors) . As well as critical apparatus elements such as `<choice>` and `<rdg>` which are not yet concepts available in any graphical music editor that I am currently aware of.

In terms of interchange with specialized repertories, Josquin Research Project scores entered in Finale and exported via MusicXML, long (and maxima) notes were not possible, and these were typically hacked visually to be half notes in the data with the notehead changed to  a breve (which had to be fixed after conversion of MusicXML data export).  Likewise for SibMEI/CRIM there is a convention of various articulations that are not used in the Renaissance that are transformed for final MEI file creation (although this is not exactly MEI Basic scope).    For the Chopin Institute use of MuseScore, I set up a system of adding text to the score in a special format that inserts Humdrum specific codes that are outside of MusicXML's concepts (so there is less cleanup of the MusicXML data when converting to Humdrum).  And there is a fair amount of post-conversion editing to add features that cannot be transported through MusicXML (such as beamSpans and tupletSpans).

And/or MEI Basic could be the minimal set of MEI that a music editor should implement for a basic MEI importer/exporter (to get the MEI seal of approval :-).  This would be more limited in scope than something such as the export capabilities of the MuseScore MEI exporter.

**Reply by ahankinson** [reply]
> it seems that anything that is capable of the MEI exporter in MuseScore would be the baseline for MEI Basic features. (and preferably import back into MuseScore or into another music editor either directly through MEI or MusicXML).

At a practical level, yes, this was actually how it was developed in the last few months of MEI Basic development. A feature that could not be expressed in MEI Basic that was available in MuseScore was added, or removed, or changed. Given that we were under a fairly tight time schedule to get something out, not every corner of MuseScore was explored. (Apologies to all the triangle players or un-pitched instrumentalists out there.)

The hope was that exactly [these](https://github.com/musescore/MuseScore/issues/22519) kinds of issues would flag places where attention was needed, and fixed in another round of development to improve both MEI Basic and MuseScore.

> Likewise for SibMEI/CRIM there is a convention of various articulations that are not used in the Renaissance that are transformed for final MEI file creation... 

SibMEI always resisted specific projects' needs. There was always pressure to put in project-specific features, but the hope was to keep it generic so that it could work across many different projects. That plugin is about at the limit of what you can do with Sibelius, notwithstanding the fabulous work currently being done by @th-we and co. 

> I would presume that things like `<beamSpan>` and `<tupletSpan>`  would be excluded...

For those who are interested... This file gives a list of the elements and attributes in MEI Basic that are supported in Verovio / MuseScore. 

https://github.com/rism-digital/verovio/blob/develop/libmei/dist/meibasic.h

I didn't see a "nicer" representation of this list anywhere, but maybe that's OK for now?
